services/auth/token/src/lambdas/generateToken.ts

```diff
@@ -16,194 +15,125 @@ interface SecretsConfig {
   name: string;
   keyName: string;
 }
-
+interface GrantValueTypes {
+  secretsConfig: SecretsConfig;
+  token: string;
+}
 interface AuthLambdaRequest {
   grant_type: string;
-  refresh_token?: string;
-  code?: string;
-}
-
-interface JWTToken {
-  personalNumber: string;
-}
-
-interface SDKError {
+  refresh_token: string;
   code: string;
-  message: string;
 }
-export async function main(event: { body: string }, context: { awsRequestId: string }) {
-  const [parseJsonError, parsedJson] = await to<AuthLambdaRequest | undefined>(
-    parseJson(event.body)
-  );
-
-  if (parseJsonError) {
-    log.warn(
-      'JSON Parse error',
-      context.awsRequestId,
-      'service-auth-token-generateToken-001',
-      parseJsonError
-    );
-
-    return response.failure(parseJsonError);
-  }
+export interface LambdaContext {
+  getSecret: (secretName: string, secretKeyName: string) => Promise<string>;
+  verifyToken: (token: string, secret: string) => Promise<Token>;
+  signToken: (jsonToSign: Token, secret: string, expireTimeInMinutes: number) => Promise<string>;
+}
 
-  const [validationError, validatedEventBody] = await to(
-    validateEventBody(
-      parsedJson ?? {
-        grant_type: '',
-      },
-      tokenValidationSchema
-    )
-  );
+/* istanbul ignore next */
+export const main = log.wrap(async (event: { body: string }) => {
+  return await lambda(event, {
+    getSecret: secrets.get,
+    verifyToken,
+    signToken,
+  });
+});
 
-  if (validationError) {
-    log.warn(
-      'Validation error',
-      context.awsRequestId,
-      'service-auth-token-generateToken-002',
-      validationError
+export async function lambda(event: { body: string }, lambdaContext: LambdaContext) {
+  try {
+    const parsedJson = parseJson(event.body);
+    const validatedEventBody = validateEventBody(parsedJson, tokenValidationSchema);
+    const grantTypeValues = getGrantTypeValues(validatedEventBody);
+
+    const decodedGrantToken = await validateToken(
+      grantTypeValues.secretsConfig,
+      grantTypeValues.token,
+      lambdaContext
     );
 
-    return response.failure(validationError);
-  }
-
-  const grantTypeValues = getGrantTypeValues(validatedEventBody);
-
-  const [validateTokenError, decodedGrantToken] = await to(
-    validateToken(grantTypeValues?.secretsConfig, grantTypeValues?.token ?? '')
-  );
-
-  if (validateTokenError) {
-    log.warn(
-      'Validate token error',
-      context.awsRequestId,
-      'service-auth-token-generateToken-003',
-      validateTokenError
+    const accessToken = await generateToken(
```

Comment:
**OPTIONAL**

Since the creation of `refreshToken` is not dependent on `accessToken`, we could just put those requests in a `Promise.all` like:

```js
    const [accessToken, refreshToken] = await Promise.all([
      generateToken(
        CONFIG_AUTH_SECRETS.accessToken,
        decodedGrantToken.personalNumber,
        ACCESS_TOKEN_EXPIRES_IN_MINUTES,
        lambdaContext
      ),
      generateToken(
        CONFIG_AUTH_SECRETS.refreshToken,
        decodedGrantToken.personalNumber,
        REFRESH_TOKEN_EXPIRES_IN_MINUTES,
        lambdaContext
      )
    ]);
```

Reply:
... and even:

```js

const generateTokenBySecretConfig = (config) => generateToken(
  config, 
  decodedGrantToken.personalNumber, 
  ACCESS_TOKEN_EXPIRES_IN_MINUTES, 
  lambdaContext 
);

const [accessToken, refreshToken] = await Promise.all([ 
  generateTokenBySecretConfig(CONFIG_AUTH_SECRETS.accessToken), 
  generateTokenBySecretConfig(CONFIG_AUTH_SECRETS.refreshToken), 
]);
```